pkg/configv2/migrate.go

```diff
@@ -0,0 +1,338 @@
+						Disk:   in.Node.Compute.Capacity.DefaultJobResourceLimits.Disk,
+						GPU:    in.Node.Compute.Capacity.DefaultJobResourceLimits.GPU,
+					},
+					Publisher: types.DefaultPublisherConfig{
+						Type: in.Node.Requester.DefaultPublisher,
+					},
+					Timeouts: types.TaskTimeoutConfig{
+						ExecutionTimeout: types.Duration(in.Node.Compute.JobTimeouts.MaxJobExecutionTimeout),
+					},
+				},
+			},
+			Ops: types.JobDefaultsConfig{
+				// TODO(forrest) [review]: Does a zero publisher value imply it has no priority or that it's not configured?
+				// Priority: 100,
+				Task: types.TaskDefaultConfig{
+					Resources: types.Resource{
+						CPU:    in.Node.Compute.Capacity.DefaultJobResourceLimits.CPU,
+						Memory: in.Node.Compute.Capacity.DefaultJobResourceLimits.Memory,
+						Disk:   in.Node.Compute.Capacity.DefaultJobResourceLimits.Disk,
+						GPU:    in.Node.Compute.Capacity.DefaultJobResourceLimits.GPU,
+					},
+					Publisher: types.DefaultPublisherConfig{
+						Type: in.Node.Requester.DefaultPublisher,
+					},
+					Timeouts: types.TaskTimeoutConfig{
+						ExecutionTimeout: types.Duration(in.Node.Compute.JobTimeouts.MaxJobExecutionTimeout),
+					},
+				},
+			},
+		},
+		// TODO(forrest) [review]: currently both the compute and requester have a job selection policy
+		// it is not clear whose policy should be migrated here.
+		JobAdmissionControl: types.JobAdmissionControl{
+			RejectStatelessJobs: in.Node.Requester.JobSelectionPolicy.RejectStatelessJobs,
+			AcceptNetworkedJobs: in.Node.Requester.JobSelectionPolicy.AcceptNetworkedJobs,
+			ProbeHTTP:           in.Node.Requester.JobSelectionPolicy.ProbeHTTP,
+			ProbeExec:           in.Node.Requester.JobSelectionPolicy.ProbeExec,
+		},
+		Logging: types.Logging{
+			Level:                "info",
+			Mode:                 string(in.Node.LoggingMode),
+			LogDebugInfoInterval: types.Duration(in.Node.Compute.Logging.LogRunningExecutionsInterval),
+		},
+		UpdateConfig: types.UpdateConfig{
+			Interval: types.Duration(in.Update.CheckFrequency),
+		},
+		FeatureFlags: types.FeatureFlags{
+			ExecTranslation: in.Node.Requester.TranslationEnabled,
+		},
+	}
+	return out, nil
+}
+
+func migrateDownloadConfig(in v1types.NodeConfig) map[string]map[string]string {
+	out := make(map[string]map[string]string)
+	if in.IPFS.Connect != "" {
+		// TODO(forrest) [review]: unsure what name to key here
+		out[models.StorageSourceIPFS] = map[string]string{
+			"endpoint": in.IPFS.Connect,
+		}
+	}
+	// TODO(forrest) [review]: assuredly the presence of these keys in the map indicate
+	// they are enable. I think we want to enable these by default, but at the same time
+	// we don't want to use, for example, the S3 downloader if it's not configured correctly.
+	// TODO(forrest) [review]: unsure what name to key here
+	out[models.StorageSourceS3] = make(map[string]string)
+	// TODO(forrest) [review]: unsure what name to key here
+	out[models.StorageSourceURL] = make(map[string]string)
```

Comment:
disregard these TODOs. The answer here is to not set them, allow them to take on the default values defined in the client code.